CHANGELOG.md

```diff
@@ -3,6 +3,191 @@
+- The `getApplyFilterFnV7` in `GridFilterOperator` was renamed to `getApplyFilterFn`.
+  If you use `getApplyFilterFnV7` directly - rename it to `getApplyFilterFn`.
+- The signature of the function returned by `getApplyFilterFn` has changed for performance reasons:
+
+```diff
+ const getApplyFilterFn: GetApplyFilterFn<any, unknown> = (filterItem) => {
+   if (!filterItem.value) {
+     return null;
+   }
+   const filterRegex = new RegExp(escapeRegExp(filterItem.value), 'i');
+-  return (cellParams) => {
+-  const { value } = cellParams;
++  return (value, row, colDef, apiRef) => {
+     return value != null ? filterRegex.test(String(value)) : false;
+   };
+ }
+```
+
+- The `getApplyQuickFilterFnV7` in `GridColDef` was renamed to `getApplyQuickFilterFn`.
+  If you use `getApplyQuickFilterFnV7` directly - rename it to `getApplyQuickFilterFn`.
+- The signature of the function returned by `getApplyQuickFilterFn` has changed for performance reasons:
+
+```diff
+ const getGridStringQuickFilterFn: GetApplyQuickFilterFn<any, unknown> = (value) => {
+   if (!value) {
+     return null;
+   }
+   const filterRegex = new RegExp(escapeRegExp(value), 'i');
+-  return (cellParams) => {
+-    const { formattedValue } = cellParams;
++  return (value, row, column, apiRef) => {
++    let formattedValue = apiRef.current.getRowFormattedValue(row, column);
+     return formattedValue != null ? filterRegex.test(formattedValue.toString()) : false;
+   };
+ };
+```
+
+#### `@mui/x-data-grid@7.0.0-alpha.0`
+
+- [DataGrid] Fix for error thrown when removing skeleton rows, after sorting is applied (#10807) @benjaminbialy
+- [DataGrid] Fix: `undefined` slot value (#10937) @romgrk
+- [DataGrid] Print selected rows by default (#10846) @cherniavskii
+- [DataGrid] Remove deprecated `components` and `componentsProps` (#10911) @MBilalShafi
+- [DataGrid] Remove legacy filtering API (#10897) @cherniavskii
+- [DataGrid] Fix keyboard navigation for actions cell with disabled buttons (#10882) @michelengelen
+- [DataGrid] Added a recipe for using non-native select in filter panel (#10916) @michelengelen
+- [DataGrid] Added a recipe to style cells without impacting the aggregation cells (#10913) @michelengelen
+- [l10n] Improve Czech (cs-CZ) locale (#10949) @luborepka
+
+#### `@mui/x-data-grid-pro@7.0.0-alpha.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-data-grid@7.0.0-alpha.0`, plus:
+
+- [DataGridPro] Autosize Columns - Fix headers being cut off (#10666) @gitstart
+- [DataGridPro] Add data source interface and basic documentation (#10543) @MBilalShafi
+
+#### `@mui/x-data-grid-premium@7.0.0-alpha.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')
+
+Same changes as in `@mui/x-data-grid-pro@7.0.0-alpha.0`, plus:
+
+- [DataGridPremium] Render aggregation label when `renderHeader` is used (#10936) @cherniavskii
+
+### Date Pickers
+
```

Comment:
I think we can add the breaking changes against #10700 here, someone from @mui/explore team, kindly check if the wording is correct?
```suggestion

#### Breaking changes

- The props `components` and `componentsProps` have been removed. Use `slots` and `slotProps` instead.

```

Reply:
@flaviendelangle / @LukasTy gentle ping! 🔔

Reply:
I've updated this suggestion and provided a tiny suggestion on the grid part as well. 🙈